common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -328,11 +325,11 @@ protected void reportInstancesLeak(String resourceType) {
     @SuppressWarnings("deprecation")
     private final class DefaultResourceLeak extends PhantomReference<Object> implements ResourceLeakTracker<T>,
```

Comment:
Since you aren't doing any cleanup here, why not use a WeakReference?

Reply:
@carl-mastrangelo good question... Let me think about it and may do another PR later on. I think this is not related to this PR.